valhalla/sif/dynamiccost.h

```diff
@@ -849,21 +850,25 @@ class DynamicCost {
 
     if (node->drive_on_right()) {
       // Did we make a uturn on a short, internal edge or did we make a uturn at a node.
-      if (has_reverse ||
-          (penalize_internal_uturns && internal_turn == InternalTurn::kLeftTurn && has_left))
+      if (has_reverse && !edge->name_consistency(idx)) {
+        seconds += kTCNameInconsistentUturn;
```

Comment:
This is the only change: add another condition here at the top: if it's a reverse turn but the edges have different names, don't apply the 10 minute penalty.